File: lintcode/cpp/Subarray_Sum.cpp
class Solution {
public:
    /**
     * @param nums: A list of integers
     * @return: A list of integers includes the index of the first number 
     *          and the index of the last number
     */
    vector<int> subarraySum(vector<int> nums){
        if(nums.empty()) return {-1,-1};
        
        for(int i = 0; i < nums.size(); i++){
            int sum = 0;
            for(int j = i; j < nums.size(); j++){
                sum += nums[j];
                if(sum == 0) return {i, j};
            }
        }
        
        return {-1,-1};
    }
};

File: lintcode/java/SubarraySum.java
public class Solution {
    /**
     * @param nums: A list of integers
     * @return: A list of integers includes the index of the first number 
     *          and the index of the last number
     */
    public List<Integer> subarraySum(int[] A) {
        if(A.length == 0) throw new IllegalArgumentException("input cannot be empty.");
        
        int sum = 0;
        for(int i = 0; i < A.length; i++){
            sum = 0;
            for(int j = i; j < A.length; j++){
                sum += A[j];
                if(sum == 0) return Arrays.asList(i, j);
            }
        }
        return Collections.emptyList();
    }
}
